jboss-kie-smartrouter/added/launch/jboss-kie-smartrouter.sh

```diff
@@ -87,31 +87,24 @@ function configure_router_location {
     local routerUrlExternal="${KIE_SERVER_ROUTER_URL_EXTERNAL}"
     local defaultInsecureHost="${HOSTNAME_HTTP:-${HOSTNAME:-localhost}}"
     local defaultSecureHost="${HOSTNAME_HTTPS:-${defaultInsecureHost}}"
-
     if [ "${host}" = "" ]; then
         host=$(find_env "${routeService}_SERVICE_HOST")
     fi
     JBOSS_KIE_ARGS="${JBOSS_KIE_ARGS} -Dorg.kie.server.router.host=${host}"
     if [ "${port}" = "" ]; then
         port=$(find_env "${routeService}_SERVICE_PORT" "9000")
     fi
-
     JBOSS_KIE_ARGS="${JBOSS_KIE_ARGS} -Dorg.kie.server.router.port=${port}"
     if [ -z "${routerUrlExternal}" ]; then
        if [ -n "${routeName}" ]; then
             if [ "${protocol}" = "https" ]; then
-                routeName="${routeName}"
-                host="${host:-${defaultSecureHost}}"
+                routeName="secure-${routeName}"
```

Comment:
see https://github.com/jboss-container-images/jboss-kie-modules/commit/6b90b305106f10baea729c4e67687bfa3f694c92

Reply:
@spolti Yes ... I had seen this with David last week but we add this "prefix" in the templates. See: https://github.com/msmagnanijr/rhpam-7-openshift-image/blob/7.4.x/templates/rhpam74-prod.yaml#L878 Also this same "idea" is used to find the routes in KieServer. See: https://github.com/jboss-container-images/jboss-kie-modules/blob/master/jboss-kie-kieserver/added/launch/jboss-kie-kieserver.sh#L439

I've done a lot of testing manually and this seems like the only way to find the routes.

Reply:
The route name is not hardcoded anymore, this secure prefix is only present on templates, not on apb and operator, the route name should now be set by this env: https://github.com/jboss-container-images/jboss-kie-modules/blob/master/jboss-kie-smartrouter/added/launch/jboss-kie-smartrouter.sh#L82

You are also re-introducing a issue previously fixed by changing this:

`	        local routeHost=$(query_route_host "${routeName}" "${host}:${port}")
	        routerUrlExternal="${protocol}://${routeHost}"

To

            routerUrlExternal=$(build_route_url "${routeName}" "${protocol}" "${host}" "${port}")`


in this case, if for some reason the routeName fail to query k8s api it will fallback on the default value which will probably be the pod hostname/ip:port.
As it is referring the smartrouter direct ipaddress/hostname the port must be 9000 otherwise the business central will fail to connect on the kieserver.

Reply:
@spolti I understand about using the variable "KIE_SERVER_ROUTER_ROUTE_NAME" but it will always have the value "${APPLICATION_NAME}-smartrouter"  but for example if i run a "oc get routes" I will have for example "myapp-smartrouter" and "secure-myapp-smartrouter".  If you use the default value of variable "KIE_SERVER_ROUTER_ROUTE_NAME" you will not be able to find the route "secure-myapp-smartrouter-rhpam-2267.apps.mmagnani.lab". That makes sense?

Reply:
https://github.com/jboss-container-images/rhpam-7-openshift-image/blob/master/templates/rhpam75-prod.yaml#L1146

Reply:
again, you are only considering templates. did you test it on operator?
A way to make it consistent across template/apb/operator would be by instructing the user what they have to do if they want to enable ssl with smartrouter.